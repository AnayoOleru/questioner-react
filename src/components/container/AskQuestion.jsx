import React, { Component, Fragment } from 'react';
import Header from '../presentation/Header';
import Footer from '../presentation/Footer';

class AskQuestion extends Component {
  render() {
    return (
      <Fragment>
        <Header />
        <div className='AskQuestion'>
          <div className='ask-question-area'>
            <div className='container'>
              <div className='meetup-title-area'>
                <div className='meetup-title'>
                    NIGERIA MODERN EXCEL & POWER BI USER GROUP
                </div>
              </div>
              <div className='ask-question-form-area'>
                <h3 className='ask-question-form-text'>Ask A Question</h3>
                <form>
                  <div className='ask-question-input-section'>
                    <input
                      type='text'
                      placeholder='Title'
                      required
                    />
                  </div>
                  <div className='ask-question-input-section'>
                    <textarea required placeholder='Body' l />
                  </div>
                  <div className='ask-question-form-submit'>
                    <button type='submit'>Submit</button>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
        <Footer />
      </Fragment>
    );
  }
}

export default AskQuestion;
