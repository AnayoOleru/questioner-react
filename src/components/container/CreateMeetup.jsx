import React, { Component, Fragment } from 'react';
import Header from '../presentation/Header';
import Footer from '../presentation/Footer';

class CreateMeetup extends Component {
  render() {
    return (
      <Fragment>
        <Header />
        <div className='CreateMeetup'>
          <div className='create-meetup-area'>
            <div className='container'>
              <div className='create-meetup-form-area'>
                <h3 className='create-meetup-form-text'>Create Meetup</h3>
                <form>
                  <div className='create-meetup-input-section'>
                    <input
                      type='text'
                      placeholder='Name'
                      required
                    />
                  </div>
                  <div className='create-meetup-input-section'>
                    <textarea required placeholder='Description' l />
                  </div>
                  <div className='create-meetup-form-submit'>
                    <button type='submit'>Submit</button>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
        <Footer />
      </Fragment>
    );
  }
}
export default CreateMeetup;
