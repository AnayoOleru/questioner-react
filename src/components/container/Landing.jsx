import React, { Component, Fragment } from 'react';
import Logo from '../../assets/images/question-mark-2-xxl.png';
import Button from '../presentation/Button';
import Header from '../presentation/Header';
import Footer from '../presentation/Footer';

class LandingPage extends Component {
  render() {
    return (
      <Fragment>
        <Header />
        <div className='LandingPage'>
          <div className='landing-area'>
            <div className='landing-background'>
              <div className='container mar-t-0'>
                <div className='landing-content-area'>
                  <img
                    className='landing-content-logo'
                    src={Logo}
                    alt='background'
                  />
                  <h1 className='landing-content-title'>Questioner</h1>
                  <h3 className='landing-content-description'>
                        Questioner helps the meetup organizer prioritize questions to be answered
                  </h3>
                  <a href='./signup.html'>
                    <Button
                      type='button'
                      id='signup'
                    >
                      Sign Up
                    </Button>
                  </a>
                </div>
              </div>
            </div>
            <div className='landing-how-it-works'>
              <div className='container'>
                <h3>How it works</h3>
                <div className='landing-how-it-works-icons'>
                  <img
                    src='https://geekdanitek.github.io/andela_fellowship_cycle_40/UI/assets/images/1141913.svg'
                    alt='landing icon'
                  />
                  <p>View list of meetups</p>
                </div>
                <div className='landing-how-it-works-icons'>
                  <img
                    src='https://geekdanitek.github.io/andela_fellowship_cycle_40/UI/assets/images/ask.png'
                    alt='landing icon'
                  />
                  <p>Ask questions on meetups</p>
                </div>
                <div className='landing-how-it-works-icons'>
                  <img
                    src='https://geekdanitek.github.io/andela_fellowship_cycle_40/UI/assets/images/vote.png'
                    alt='landing icon'
                  />
                  <p>
                      See votes as they bubble to the top or bottom of the log
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <Footer />
      </Fragment>
    );
  }
}
export default LandingPage;
