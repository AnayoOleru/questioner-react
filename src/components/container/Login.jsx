import React, { Component, Fragment } from 'react';
import { Link } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSpinner } from '@fortawesome/free-solid-svg-icons';
import { connect } from 'react-redux';
import { login } from '../../actions';
import jsonFormatter from '../../utils/jsonFormat.utils';

class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: ''
    };
  }

  handleChange = (e) => {
    this.setState({ [e.target.name]: e.target.value });
  }

  handleLogin = async (e) => {
    e.preventDefault();
    const { loginAction } = this.props;
    const { email, password } = this.state;
    await loginAction({
      email,
      password
    });
  }

  render() {
    const { email, password } = this.state;
    const { auth } = this.props;
    return (
      <Fragment>
        <div className='Login'>
          <div className='signin-area'>
            <h1 className='signin-title'>Questioner</h1>
            <h4 className='signin-title-description'>
              Crowd-source questions platform for meetup
            </h4>
            <div className='signin-card'>
              <h4 className='signin-card-title'>Sign-in</h4>
              <form onSubmit={this.handleLogin}>
                {
                  auth && auth.error && auth.error.error && (
                  <div
                    className='error-section'
                    dangerouslySetInnerHTML={{ __html: jsonFormatter(JSON.stringify(auth.error.error)) }}
                  />
                  )
                }
                <div className='signin-input-section'>
                  <input
                    type='email'
                    placeholder='Email'
                    onChange={this.handleChange}
                    value={email}
                    required
                    name='email'
                  />
                </div>
                <div className='signin-input-section'>
                  <input
                    type='password'
                    placeholder='Password'
                    onChange={this.handleChange}
                    value={password}
                    required
                    name='password'
                  />
                </div>
                <div className='signin-submit'>
                  <button
                    type='submit'
                    disabled={auth.isLoading}
                  >
                    { auth.isLoading ? <FontAwesomeIcon icon={faSpinner} spin size='1x' /> : 'Submit' }
                  </button>
                </div>
              </form>
              <div className='signin-card-footer'>
                <small>
                  <Link to='/signup'>New here? Sign up</Link>
                </small>
              </div>
            </div>
          </div>
        </div>
      </Fragment>
    );
  }
}

const mapStateToProps = (state) => {
  const { auth } = state;
  return { auth };
};

const mapDispatchToProps = {
  loginAction: payload => login(payload)
};

export default connect(mapStateToProps, mapDispatchToProps)(Login);
