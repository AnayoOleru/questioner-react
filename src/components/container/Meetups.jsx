import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import { HeaderComponent } from '../presentation/Header';
import Footer from '../presentation/Footer';
import MeetupCard from '../presentation/MeetupCard';
import { getMeetups } from '../../actions';
import FullPageLoader from '../presentation/FullPageLoader';

export class Meetups extends Component {
  async componentDidMount() {
    const { getMeetupsAction } = this.props;
    await getMeetupsAction();
  }

  render() {
    const { meetup } = this.props;
    return (
      <Fragment>
        <HeaderComponent />
        { meetup && meetup.isLoading && <FullPageLoader /> }
        <div className='Meetups'>
          <div className='meetups-area'>
            <div className='container'>
              {
                meetup && meetup.meetups && meetup.meetups.map(singleMeetup => (
                  <MeetupCard
                    key={singleMeetup.id}
                    topic={singleMeetup.topic}
                    createdon={singleMeetup.createdon}
                    location={singleMeetup.location}
                    happeningon={singleMeetup.happeningon}
                    id={singleMeetup.id}
                  />
                ))
              }
            </div>
          </div>
        </div>
        <Footer />
      </Fragment>
    );
  }
}

const mapStateToProps = ({ meetup }) => ({ meetup });

const mapDispatchToProps = {
  getMeetupsAction: () => getMeetups()
};
export const MeetupsComponent = connect(mapStateToProps, mapDispatchToProps)(Meetups);
