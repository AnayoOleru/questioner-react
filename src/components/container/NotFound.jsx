import React from 'react';
import { Link } from 'react-router-dom';

const NotFound = () => (
  <div>
        Not Found,
    {' '}
    <Link to='/'>Homepage</Link>
  </div>
);
export default NotFound;
