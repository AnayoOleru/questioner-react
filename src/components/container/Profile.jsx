import React, { Component, Fragment } from 'react';
import { Link } from 'react-router-dom';
import Header from '../presentation/Header';
import Footer from '../presentation/Footer';
import TopQuestionCard from '../presentation/TopQuestionCard';

class Profile extends Component {
  render() {
    return (
      <Fragment>
        <Header />
        <div className='Profile'>
          <div className='user-profile-area'>
            <div className='container'>
              <div className='meetups-link'>
                <Link to='/meetup/create'>
                  <button>
                    Create A Meetup
                  </button>
                </Link>
                <Link to='/meetups'>
                  <button>
                  View All Meetups
                  </button>
                </Link>
              </div>
              <div className='user-profile-question-post'>
                <div className='user-profile-card'>
                  <h1>24</h1>
                  <span>Questions Asked</span>
                </div>
              </div>
              <div className='user-profile-question-comment'>
                <div className='user-profile-card'>
                  <h1>13</h1>
                  <span>Questions Commented</span>
                </div>
              </div>
              <h3 className='top-question-feed-text'>Top questions feed</h3>
              <TopQuestionCard />
            </div>
          </div>
        </div>
        <Footer />
      </Fragment>
    );
  }
}
export default Profile;
