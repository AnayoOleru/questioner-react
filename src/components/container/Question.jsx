import React, { Component, Fragment } from 'react';
import Header from '../presentation/Header';
import Footer from '../presentation/Footer';
import Comment from '../presentation/Comment';

class Question extends Component {
  render() {
    return (
      <Fragment>
        <Header />
        <div className='Question'>
          <div className='question-area'>
            <div className='container'>
              <div className='meetup-title-area'>
                <div className='meetup-title'>
                    NIGERIA MODERN EXCEL & POWER BI USER GROUP
                </div>
              </div>
              <div className='question-card'>
                <div className='section'>
                  <div className='section-left'>
                    <div className='vote'>
                      <button className='vote-up' />
                      <span>2423</span>
                      <button className='vote-down' />
                    </div>
                  </div>
                  <div className='section-right'>
                    <h2 className='question-card-title'>
                        What is the purpose of this meetup like seriously i dont understand.
                    </h2>
                    <div className='question-card-body-content'>
                      <div className='question-card-description'>
                        <p>
                        Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod
                        tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
                        quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
                        consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
                        cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
                        proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
                        </p>
                      </div>
                      <span className='question-card-comments'>
                        13 comments,
                        {' '}
                        <b>14th, Dec 2018</b>
                      </span>
                    </div>
                  </div>
                </div>
                <div className='question-card-comment-area'>
                  <form>
                    <div className='question-card-input-section'>
                      <textarea
                        placeholder='Comment here'
                        required
                      />
                    </div>
                    <div className='question-card-button-area'>
                      <button>Submit</button>
                    </div>
                  </form>
                </div>
                <div className='question-card-footer'>
                  <small>The Conoil group</small>
                </div>
              </div>
              <Comment />
              <Comment />
            </div>
          </div>
        </div>
        <Footer />
      </Fragment>
    );
  }
}

export default Question;
