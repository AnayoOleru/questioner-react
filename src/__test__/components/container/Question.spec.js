import React from 'react';
import { shallow } from 'enzyme';
import { Question } from '../../../components/container/Question';

describe('Question Component', () => {
  let wrapper;
  beforeEach(() => {
    const props = {
      match: {
        params: {
          id: '1'
        }
      },
      getQuestionAction: jest.fn(),
      question: {
        question: {
          title: 'hello me',
          createdon: '23 May, 2019',
          body: 'Heleeef fefefe f',
          happeningon: '23 May, 2019',
          id: 'Hedede'
        },
        isLoading: false
      },
      comment: {
        comments: [{
          createdon: '23 May, 2019',
          firstname: 'Daniel',
          lastname: 'Geek',
          comment: 'lold dsdssd'
        }],
        isLoading: false

      },
    };
    wrapper = shallow(<Question {...props} />);
  });
  test('Question component should render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
