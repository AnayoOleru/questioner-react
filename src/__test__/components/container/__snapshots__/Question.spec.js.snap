// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question Component Question component should render 1`] = `ShallowWrapper {}`;
